plugins/catalog-backend/src/service/AuthorizedCatalogProcessingOrchestrator.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { NotAllowedError } from '@backstage/errors';
+import { catalogEntityValidatePermission } from '@backstage/plugin-catalog-common/alpha';
+import { AuthorizeResult } from '@backstage/plugin-permission-common';
+import {
+  CatalogProcessingOrchestrator,
+  EntityProcessingRequest,
+  EntityProcessingResult,
+} from '../processing/types';
+import {
+  BackstageCredentials,
+  PermissionsService,
+} from '@backstage/backend-plugin-api';
+
+export class AuthorizedCatalogProcessingOrchestrator
```

Comment:
I think that we should convert this into a `ValidationService` class instead which is called directly by the router to validate with a `CatalogProcessingOrchestrator` instead. 

It feels a little odd to add the `BackstageCredentials` argument to the `Orchestrator` which shouldn't really have any knowledge of that, its designed to be used with the processing loop and not aware of incoming http requests or whatnot.

Reply:
I chose this approach to maintain consistency with how permissions are handled across the catalog. For example, [`EntitiesCatalog`](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/service/AuthorizedEntitiesCatalog.ts#L111), [`LocationService`](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/service/AuthorizedLocationService.ts#L60), [`RefreshService`](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/service/AuthorizedRefreshService.ts#L44) all have `BackstageCredentials` as an argument. I don't see why the `Orchestrator` would be any different in this regard.

Reply:
Semantically I don't think it makes that much sense. You mentioned also the `LocationService` and the `RefreshService` which is the pattern I'm trying to recommend here too, the `Orchestrator` is used in a lot more places too than just the router. Just feeling that this should be taken care of at the Routing level as much as possible. Let's see how it feels?

Reply:
@benjdlambert changed it so that permission checking is handled at the router level, lmk what you think.